moto/ec2/models.py

```diff
@@ -3011,11 +3016,26 @@ def get_filter_value(self, filter_name):
         else:
             return super(VPC, self).get_filter_value(filter_name, "DescribeVpcs")
 
+    def modify_vpc_tenancy(self, tenancy):
+        if tenancy == "default":
+            self.instance_tenancy = tenancy
+        else:
+            raise UnsupportedTenancy(tenancy)
+        return True
+
     def associate_vpc_cidr_block(
         self, cidr_block, amazon_provided_ipv6_cidr_block=False
     ):
         max_associations = 5 if not amazon_provided_ipv6_cidr_block else 1
 
+        for cidr in self.cidr_block_association_set.copy():
```

Comment:
Note for @bblommers : This is done to prevent error messages where unused association sets (in state `disassociated`) exist and take up all slots. We've verified this behavior against real AWS. For example, if 1 IPv6 CIDR block association already exists in state `disassociated`, we should still allow associating a new CIDR block here, hence `pop`'ing it from line 3038 below.